moto/iam/models.py

```diff
@@ -569,6 +569,16 @@ def get_role(self, role_name):
                 return role
         raise IAMNotFoundException("Role {0} not found".format(role_name))
 
+    def delete_role(self, role_name):
```

Comment:
I've seen the implementation of `delete_policy()` and it seems slightly simpler, what about:

```python
    def delete_role(self, role_name):
        if role_name not in self.roles:
            raise IAMNotFoundException(
                "Role {0} not found".format(role_name))

        del self.policies[role_name]
```